driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -1448,119 +1448,125 @@ private Manager(
     // on it so synchronized is good enough.
     synchronized void init() {
       checkNotClosed(this);
-      if (isInit) return;
+      if (isInit) {
+        if (initException == null) {
+          return;
+        }
+        throw new IllegalStateException(
```

Comment:
We should only end up here if initialization was attempted, and did not complete. This method is synchronized so we don't need to worry about a race condition.